### DISCUSSION TITLE
Coverage issue in vue files when upgraded with version 1.1.3  or 1.2.0

### DISCUSSION THREAD

**Discussion by roshnimridha04** [vitest-dev/vitest#4991] [General]
After I have upgraded to latest, Iam seeing weird issues in coverage. Before the upgrade it was all fine. For some files its still shows the coverage fine. but for some if I move the script after my template as shown below then it shows correctly else it shows not get covered:
/* v8 ignore start */
<template>
 my template here
</template>
/* v8 ignore stop */

<script setup lang="ts">

//my script here
</script>

Please let me know why is this happening after the upgrade.

**Comment by AriPerkkio** [comment]
Did you update anything else than Vitest? Check if your lockfile contains changes to packages like Vue Vite plugin, vue loader or similar.

**Reply by roshnimridha04** [reply]
Earlier I guess it was c8 for coverage which used to work well. With v8 we are facing the issue.
Using the following version of these

 "@types/jsdom": "^21.1.6",
    "@types/node": "^20.11.4",
    "@vitejs/plugin-vue": "^5.0.3",
    "@vitest/coverage-v8": "^1.2.0",
    "jsdom": "^23.2.0",
    "vite": "^5.0.11",
    "vitest": "^1.2.0",

**Reply by roshnimridha04** [reply]
This is the vue version that we are using
"vue": "^3.4.14",

And the config looks like this:
test: {
    globals: true,
    environment: "jsdom",
    clearMocks: true,
    mockReset: true,
    setupFiles: ["tests/unit/setup.ts"],
    coverage: {
      reporter: ["lcov", "text", "text-summary"],
      enabled: true,
      all: true,
      thresholds: {
        lines: 80,
        functions: 80,
        branches: 80,
        statements: 80
      },
      extension: [".vue", ".ts", ".js"],
      exclude: [...configDefaults.exclude]
    },
    exclude: [...configDefaults.exclude]
  }
@AriPerkkio

**Comment by roshnimridha04** [comment]
Few more information that I can provide after investigating is our code coverage was working all fine when we used "@vitejs/plugin-vue": "^4.5.2",
 "@vitest/coverage-v8": "^1.0.4", with vitest 1.0.4 

After that we directly upgraded to vitest  and v8-coverage to 1.1.3 and "@vitejs/plugin-vue": "^5.0.2"  and here onwards the coverage started showing weird issues.

**Reply by AriPerkkio** [reply]
How about versions `1.1.0`, `1.1.1` and `1.1.2`?

**Reply by roshnimridha04** [reply]
I tried these versions with @vitejs/plugin-vue: ^5.0.2 and these have same issues

**Reply by AriPerkkio** [reply]
So just to confirm: `1.0.4` is working OK but `1.1.0` breaks?

**Reply by roshnimridha04** [reply]
Yes

**Reply by AriPerkkio** [reply]
We didn't do anything coverage related on `1.1.0` so this sounds weird. Please provide runnable minimal reproduction so I can take a look at this.

**Reply by roshnimridha04** [reply]
HI @AriPerkkio , So this is what I have observed. I have latest version of vitest and when I use "@vitejs/plugin-vue": "4.5.2" with it it works.  With @vitejs/plugin-vue 4.6.0 onwards it fails

**Reply by AriPerkkio** [reply]
And does `@vitejs/plugin-vue@4.6.0` work with `vitest@1.0.4`?

**Reply by roshnimridha04** [reply]
@AriPerkkio  @vitejs/plugin-vue@4.6.0  does not work with vitest@1.0.4

**Reply by AriPerkkio** [reply]
Ok, it sounds like this is definitely an issue on `@vitejs/plugin-vue` side. I would recommend to open a bug report there and ask the maintainers to take a look at changes in source maps between `@vitejs/plugin-vue@4.5.2` and `@vitejs/plugin-vue@4.6.0`.
